### PR TITLE
Query and Update animal table. 

### PR DESCRIPTION
**Pull request by Muhammad0602**
### For this milestone of the project I have implemented the following: 
- Added a new column `species` to my table
- Inserted new values into the table
- Inside a transaction updated the `animals` table 
 here is the screenshot: 
image
Inside a transaction Updated the `animals` table by setting the species column to:
- `digimon` for all animals that have a name ending in mon.
 - `pokemon` for all animals that don't have species already set.
 
Screenshot (78)
- Inside a transaction deletedall records in the `animals` table, then rolled back the transaction.
image
-Inside a transaction:
Deleted all animals born after Jan 1st, 2022.
Created a savepoint for the transaction.
Updated all animals' weight to be their weight multiplied by -1.
Rolledback to the savepoint
Updated all animals' weights that are negative to be their weight multiplied by -1.
Commited transaction
image
-Wrote queries to answer the following questions:
How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
image
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -3,3 +3,13 @@
...
+
+INSERT INTO animals  (id, name, date_of_birth, escape_attempts, neutered, weight_kg)
+OVERRIDING SYSTEM VALUE
+  VALUES (5, 'Charmander', '2020-02-08', 0, false, 11), 
+  (6, 'Plantmon', '2021-11-15', 2, true, 5.7), 
+  (7, 'Squirtle', '1993-04-02', 3, false, 12.13),
+  (8, 'Angemon', '2005-06-12', 1, true, 45),
+  (9, 'Boarmon', '2005-06-07', 7, true, 20.4),
+  (10, 'Blossom', '1998-10-13', 3, true, 17),
+  (11, 'Ditto', '2022-05-14', 4, true, 22);
```
Comment (alexander16108) [inline data.sql]: please input the correct values for the weight as some of the inputs ought to be negative numbers

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,48 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
+
```
Comment (alexander16108) [inline queries.sql]: please as required you are asked to check if the changes were made before and after the rollback

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,48 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SET species = 'pokemon'
+WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+
```
Comment (alexander16108) [inline queries.sql]: please as required you are asked to check if the changes were made before and after the rollback

File: queries.sql [hunk 4]
```diff
@@ -8,3 +8,48 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, AVG(escape_attempts) AS avg_escape_attempts FROM animals 
+GROUP BY neutered;
+
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts FROM animals
+WHERE data_of_birth BETWEEN '1990-01-01' AND '2000-12-30'
+GROUP BY species;
```
Comment (alexander16108) [inline queries.sql:59]: please update your queries with the correct values and try again

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Muhammad0602 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- good pr
- add screenshots

## Required Changes ♻️
_Check the comments under the review._

please fix the broken link on your readme file your mit license doesn't not work

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (Muhammad0602) [reply data.sql]: Hello @alexander16108 thanks for the feedback. I noticed this mistake and corrected it in the database as you can see in screenshots, but I forgot to change it on data.sql.

File: queries.sql (hunk 4 above)
Comment (Muhammad0602) [reply queries.sql:59]: and as I said the queries were done with the correct values, you can notice it via screenshots. So, I think I just have to change the files in VSCode. Thanks very much for your kind feedback!

**Review by osamaashraf6 — CHANGES_REQUESTED** [review]
Hi @Muhammad0602 😊, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
SoCloseAlmostGIF (3)

## Highlights🎉🎉

- [x] Add screenshots💯
- [x] Good descriptive pull request👌

## Required Changes ♻️
- [x] Kindly 😊  as stated by the previous reviewer, please fix the broken link on your readme file your mit license doesn't work😃 



## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Muhammad0602** [comment]
Hello @osamaashraf6, thanks I did not notice that.😁

**Review by amejid — APPROVED** [review]
Hi @Muhammad0602, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Correct workflow :heavy_check_mark: 
- Queries with correct results :heavy_check_mark: 
- SQL in different files :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**